components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityGovernanceServiceImpl.java

```diff
@@ -233,10 +237,68 @@ public ConnectorConfig getConnectorWithConfigs(String tenantDomain,
+                }
                 return connectorConfig;
             }
         }
         return null;
     }
 
+    /**
+     * This method is used to make sure property value useNumericCharacters and useAlphanumericCharacters both uses
+     * same user input.
+     *
+     * @param configurationDetails Configuration details of the email OTP connector.
+     */
+    private void convertPropertyUseNumericToUseAlphaNumeric(Map<String, String> configurationDetails) {
+
+        if (configurationDetails.containsKey(EMAIL_OTP_USE_ALPHANUMERIC_CHARS) &&
+                configurationDetails.containsKey(EMAIL_OTP_USE_NUMERIC_CHARS)) {
+            boolean useNumericChars = !Boolean.parseBoolean(configurationDetails.get(EMAIL_OTP_USE_ALPHANUMERIC_CHARS));
+            configurationDetails.put(EMAIL_OTP_USE_NUMERIC_CHARS, String.valueOf(useNumericChars));
+        }
+    }
+
+    /**
+     * This method is used to convert the property value useNumericCharacters to useAlphanumericCharacters.
+     *
+     * @param connectorName   Name of the connector.
+     * @param connectorConfig Connector configuration.
+     */
+    private void convertPropertyUseNumericToUseAlphaNumeric(String connectorName, ConnectorConfig connectorConfig) {
+
+        // Verify the order of the connector properties hasn't changed.
+        if (EMAIL_OTP_USE_ALPHANUMERIC_CHARS.equals(connectorConfig.getProperties()[3].getName()) &&
+                EMAIL_OTP_USE_NUMERIC_CHARS.equals(connectorConfig.getProperties()[4].getName())) {
+            if (connectorConfig.getProperties()[4].getValue() != null) {
+                // Extract the value of the useNumericCharacters property.
+                boolean useAlphanumericChars = !Boolean.parseBoolean(connectorConfig.getProperties()[4].getValue());
+                // Assign the value to the alphanumeric property.
+                connectorConfig.getProperties()[3].setValue(String.valueOf(useAlphanumericChars));
+            }
+        } else {
+            log.debug("The order of the connector properties has changed for the connector: " + connectorName);
+        }
+    }
+
+    /**
+     * This method is used to check whether the connector is email OTP connector or not.
+     *
+     * @param connectorName   Name of the connector.
+     * @param connectorConfig Connector configuration.
+     *
+     * @return True if the connector is email OTP connector and have both old and new email OTP type related configs.
+     */
+    private boolean isEmailOTPConnectorWithNewConfig(String connectorName, ConnectorConfig connectorConfig) {
+
+        /*
+        If the new config is also added, the length of the properties will be 5 with both old
+        and new OTP type properties. Here the purpose of adding 'at least check' is to allow
+        developers to safely add new configs.
+         */
+        return EMAIL_OTP_AUTHENTICATOR.equals(connectorName) && connectorConfig.getProperties().length >= 5;
```

Comment:
Need to evaluate this logic. Validating the length of the properties and deciding if it is due to new configs is error-prone.

Reply:
Resolved